File: 0383-ransom-note/0383-ransom-note.py
from collections import Counter
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        ransom_count = Counter(ransomNote)
        magazine_count = Counter(magazine)

        for letter, count in ransom_count.items():
            if magazine_count[letter] < count:
                return False
        return True        

File: 0383-ransom-note/0383-ransom-note.dart
class Solution {
  bool canConstruct(String ransomNote, String magazine) {
       Map<String, int> wordMap = {};
       for(String char in magazine.split("")) {
          wordMap[char] = (wordMap[char] ?? 0) + 1;
       }
       for(String char in ransomNote.split("")) {
          wordMap[char] = (wordMap[char] ?? 0) - 1;
          if(wordMap[char]! < 0) return false;
       }
    return true;
}}